spec/services/product_tag_rules_filterer_spec.rb

```diff
@@ -50,9 +50,98 @@
     }
     let(:filterer) { described_class.new(distributor, customer, variants_relation) }
 
-    context "when the distributor has no rules" do
-      it "returns the relation unchanged" do
-        expect(filterer.call).to eq variants_relation
+    describe "#call" do
+      before do
+        v1
+        v2
+        v3
+        v4
+      end
+
+      context "when the distributor has no rules" do
+        it "returns the relation unchanged" do
+          expect(filterer.call).to eq variants_relation
+        end
+      end
+
+      context "with hide rule" do
+        it "hides the variant matching the rule" do
+          customer.update_attribute(:tag_list, hide_rule.preferred_customer_tags)
+          variant_hidden_by_rule.update_attribute(:tag_list, hide_rule.preferred_variant_tags)
+
+          expect(filterer.call).not_to include(variant_hidden_by_rule.variant)
+        end
+
+        context "with mutiple conflicting rules" do
```

Comment:
```suggestion
        context "with multiple conflicting rules" do
```

I wonder if you have tried spell checking in your editor? ;)

Reply:
Also below